### DISCUSSION TITLE
Extending functionality - plugin method?

### DISCUSSION THREAD

**Discussion by joshabts** [mediacms-io/mediacms#734] [Q&A] [answered]
For extending functionality, is there a recommended method for "plugins" for MediaCMS?  Or would it be just modifying the models in the code and then for upgrades having to re-merge that code?  I see it is using the Django CMS framework, but not seeing the best way to create a plugin to MediaCMS.  The developer docs seem to be more around working with requests than developing against the codebase.

Thanks!

**Comment by mgogoulos** [comment, marked as answer]
Hi, MediaCMS is using Django framework and not Django CMS (which is a great CMS, built in Django). 

For totally new models maybe consider adding them as separate Django apps, so their migrations are isolated than the ones used by the project's main apps (files/users). But you are correct that any change introduced on the backend code (eg html templates, Django views) or on the frontend code, will then have to be checked carefully when merging with main branch, if you want to keep up with later updates. There is not an out-of-the-box way to do things for adding new functionality, I'd seen a number of installations that are heavily modified but I'm not sure whether they have been syncing with changes in main.
Hope this helps

**Reply by joshabts** [reply]
This does help!  I am getting more familiar with Django, so learning a bit as I go in terms of apps and how to extend the models.  I think to some extent I can go with a new app with my own models, but will likely also need to extend the core model for files.

I'll think on this a bit more and see what I can come up with!  Thanks again!
